report/src/main/kotlin/com/hartwig/actin/report/pdf/ReportContentProvider.kt

```diff
@@ -161,47 +149,55 @@ class ReportContentProvider(private val report: Report, private val enableExtend
                 report.config.includeEligibleSOCTreatmentSummary
             }, EligibleApprovedTreatmentGenerator(report, contentWidth).takeIf {
                 report.config.includeApprovedTreatmentsInSummary
-            }) +
-                primaryCohortsGenerators + otherCohortGenerators +
-                listOfNotNull(
-                    localTrialGenerator.takeIf { report.config.includeExternalTrialsInSummary },
-                    nonLocalTrialGenerator.takeIf { report.config.includeExternalTrialsInSummary },
-                    IneligibleActinTrialsGenerator.forOpenCohorts(
-                        cohorts, report.requestingHospital, contentWidth, enableExtendedMode
-                    ).takeIf {
-                        report.config.includeIneligibleTrialsInSummary
-                    })
+            }) + trialTableGenerators
     }
 
-    private fun getGeneratorsForSource(
-        cohorts: List<InterpretedCohort>,
+    private fun getGenerators(
```

Comment:
Minor but would rename to `createGenerators` as your function actually does something. "Get" typically implies a trivial lookup